### PR TITLE
Time series constraints

### PR DESCRIPTION
**Pull request by kdheepak**

### CODE REVIEW

File: src/devices_models/devices/hydro_generation.jl [hunk 1]
```diff
@@ -116,21 +121,42 @@ function add_constraints!(
 ) where {V <: PSY.HydroGen, W <: HydroDispatchRunOfRiver}
     # Use parameters here
```
Comment (jd-lara) [inline src/devices_models/devices/hydro_generation.jl:122]: remove stale comments

File: src/devices_models/device_constructors/renewablegeneration_constructor.jl [hunk 2]
```diff
@@ -27,7 +27,7 @@ function construct_device!(
 
     # Parameters
     add_parameters!(container, ActivePowerTimeSeriesParameter, devices, model)
-    add_parameters!(container, ReactivePowerTimeSeriesParameter, devices, model)
+    # add_parameters!(container, ReactivePowerTimeSeriesParameter, devices, model)
```
Comment (jd-lara) [inline src/devices_models/device_constructors/renewablegeneration_constructor.jl:30]: remove stale comments and code

File: src/devices_models/device_constructors/hydrogeneration_constructor.jl [hunk 3]
```diff
@@ -764,7 +785,8 @@ function construct_device!(
         S,
         get_feedforward(model),
     )
-    commit_hydro_active_power_ub!(container, devices, model, get_feedforward(model))
+    # TODO: check with jose if this being handled by the above add_constraints function is the right thing to do
+    # commit_hydro_active_power_ub!(container, devices, model, get_feedforward(model))
```
Comment (jd-lara) [inline src/devices_models/device_constructors/hydrogeneration_constructor.jl:789]: @sourabhdalvi can you confirm which constraints are needed for the hydro formulations?

File: test/runtests.jl [hunk 4]
```diff
@@ -53,12 +53,25 @@ include("test_utils/operations_problem_templates.jl")
 const LOG_FILE = "power-simulations-test.log"
 
 const DISABLED_TEST_FILES = [
-    "test_simulation_sequence.jl",
+    "test_basic_model_structs.jl",
```
Comment (jd-lara) [inline test/runtests.jl:56]: not all theses tests should be disabled. Only `simulation` and `services` tests should be disabled right now.

File: src/devices_models/devices/hydro_generation.jl [hunk 5]
```diff
@@ -237,14 +263,17 @@ This function define the range constraint specs for the
...
     container::OptimizationContainer,
-    devices,
+    T::Type{ActivePowerVariableLimitsConstraint},
+    U::Type{<:VariableType},
+    devices::IS.FlattenIteratorWrapper{V},
     model::DeviceModel{V, W},
+    X::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward},
 ) where {V <: PSY.HydroGen, W <: AbstractHydroUnitCommitment}
     # Use Parameters Here
```
Comment (jd-lara) [inline src/devices_models/devices/hydro_generation.jl:275]: remove comment.

File: src/devices_models/devices/common/range_constraint.jl [hunk 6]
```diff
@@ -463,3 +463,133 @@ function reserve_energy_ub!(
...
+        if multiplier[name, t] !== NaN
+            constraint[name, t] = JuMP.@constraint(
+                container.JuMPmodel,
+                expression_ub <= multiplier[name, t] * parameter[name, t]
+            )
+        end
+    end
+end
+
+function add_parameterized_upper_bound_bigM_range_constraints(
```
Comment (jd-lara) [inline src/devices_models/devices/common/range_constraint.jl:552]: this function should be commented out since it is untested and not in use.

**Review by jd-lara — CHANGES_REQUESTED** [review]
The PR looks good, there are a few things that need to be addressed before merging.

**Comment by kdheepak** [comment]
I've made the changes you've requested.

**Review by jd-lara — APPROVED** [review]

**Pull request merged**